tests/Testcontainers.Platform.Linux.Tests/ReusableResourceTest.cs

```diff
@@ -0,0 +1,80 @@
+namespace Testcontainers.Tests;
+
+public sealed class ReusableResourceTest : IAsyncLifetime, IDisposable
+{
+    private readonly DockerClient _dockerClient = TestcontainersSettings.OS.DockerEndpointAuthConfig.GetDockerClientConfiguration(Guid.NewGuid()).CreateClient();
+
+    private readonly FilterByProperty _filters = new FilterByProperty();
+
+    private readonly IList<IAsyncDisposable> _disposables = new List<IAsyncDisposable>();
+
+    private readonly string _labelKey = Guid.NewGuid().ToString("D");
+
+    private readonly string _labelValue = Guid.NewGuid().ToString("D");
+
+    public ReusableResourceTest()
+    {
+        _filters.Add("label", string.Join("=", _labelKey, _labelValue));
+    }
+
+    public async Task InitializeAsync()
+    {
+        for (var _ = 0; _ < 3; _++)
+        {
+            // We are running in a single session, we do not need to disable the cleanup feature.
+            var container = new ContainerBuilder()
+                .WithImage(CommonImages.Alpine)
+                .WithEntrypoint("sleep")
+                .WithCommand("infinity")
+                .WithLabel(_labelKey, _labelValue)
+                .WithReuse(true)
+                .Build();
+
+            var network = new NetworkBuilder()
+                .WithName(_labelKey)
+                .WithLabel(_labelKey, _labelValue)
+                .WithReuse(true)
+                .Build();
+
+            var volume = new VolumeBuilder()
+                .WithName(_labelKey)
+                .WithLabel(_labelKey, _labelValue)
+                .WithReuse(true)
+                .Build();
+
+            await Task.WhenAll(container.StartAsync(), network.CreateAsync(), volume.CreateAsync())
```

Comment:
The only thing I'm not really happy with is that _reuse_ doesn't support or might run into race conditions when creating and starting the same configuration in parallel. Since reuse shouldn't be used within the same processes, this isn't a huge issue, I guess.